idea/src/org/jetbrains/kotlin/idea/intentions/ReplaceMapGetOrDefaultIntention.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2010-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.util.TextRange
+import org.jetbrains.kotlin.idea.core.replaced
+import org.jetbrains.kotlin.idea.inspections.collections.isCalling
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.findDescendantOfType
+import org.jetbrains.kotlin.psi.psiUtil.startOffset
+
+class ReplaceMapGetOrDefaultIntention : SelfTargetingRangeIntention<KtDotQualifiedExpression>(
+    KtDotQualifiedExpression::class.java, "Replace with indexing and elvis operator"
+) {
+    companion object {
+        private val getOrDefaultFqName = FqName("kotlin.collections.Map.getOrDefault")
+    }
+
+    override fun applicabilityRange(element: KtDotQualifiedExpression): TextRange? {
+        val callExpression = element.callExpression ?: return null
+        val calleeExpression = callExpression.calleeExpression ?: return null
+        if (!callExpression.isCalling(getOrDefaultFqName)) return null
+        val (firstArg, secondArg) = callExpression.arguments() ?: return null
+        text = "Replace with ${element.receiverExpression.text}[${firstArg.text}] ?: ${secondArg.text}"
+        return calleeExpression.textRange
```

Comment:
We should also check that `Map`'s value type is not nullable, otherwise we will get a wrong result like 
```kotlin
fun test(map: Map<Int, String?>) =
  map[1] ?: "default" // Overwriting returned null value :(
```

Reply:
Fixed

Reply:
Thank you! will merge as remote run will be green ^_^